chpl/chpl-service/src/main/java/gov/healthit/chpl/caching/AsynchronousCacheInitialization.java

```diff
@@ -12,27 +12,40 @@
 import org.springframework.transaction.annotation.Transactional;
 
 import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.exception.JiraRequestFailedException;
 import gov.healthit.chpl.manager.CertificationIdManager;
 import gov.healthit.chpl.manager.CertifiedProductSearchManager;
 import gov.healthit.chpl.manager.DimensionalDataManager;
+import gov.healthit.chpl.service.DirectReviewService;
 
 @Component
 public class AsynchronousCacheInitialization {
     private static final Logger LOGGER = LogManager.getLogger(AsynchronousCacheInitialization.class);
 
-    @Autowired
     private CertificationIdManager certificationIdManager;
-    @Autowired
     private DimensionalDataManager dimensionalDataManager;
-    @Autowired
     private CertifiedProductSearchManager certifiedProductSearchManager;
+    private DirectReviewService drService;
+
+    @Autowired
+    public AsynchronousCacheInitialization(CertificationIdManager certificationIdManager,
+            DimensionalDataManager dimensionalDataManager,
+            CertifiedProductSearchManager certifiedProductSearchManager,
+            DirectReviewService drService) {
+        this.certificationIdManager = certificationIdManager;
+        this.dimensionalDataManager = dimensionalDataManager;
+        this.certifiedProductSearchManager = certifiedProductSearchManager;
+        this.drService = drService;
+    }
 
     @Async
     @Transactional
     public Future<Boolean> initializeSearchOptions() throws EntityRetrievalException {
         LOGGER.info("Starting cache initialization for SearchViewController.getPopulateSearchData()");
         dimensionalDataManager.getSearchableDimensionalData(true);
         dimensionalDataManager.getSearchableDimensionalData(false);
+        dimensionalDataManager.getDimensionalData(true);
+        dimensionalDataManager.getDimensionalData(false);
```

Comment:
Was trying to figure out why we call the same method twice with two different booleans. Don't see anything in the PR about it, so will ask here. This just looks odd...

Reply:
That's how ehCache works. It will cache the response to the getDimensionalData call with the key "true" and another entry in the appropriate cache with the key "false". Then when we call the method whether we call it with an argument of true or false the data is cached. I don't think this code change was related to the ticket but the above lines (45-46) are deprecated methods and I think we missed caching the non-deprecated methods intended to replace them. That's why I added them here.